week3/homework/Program.cs

```diff
@@ -0,0 +1,201 @@
+    }
+}
+
+
+// Create a class named ExchangeRate with a constructor with two string parameters, fromCurrency and toCurrency. 
+//Add a decimal property called Rate and method Calculate with decimal parameter amount return value of the method 
+//should be a product of Rate and amount multiplication.
+// Bonus: We should also check that Rate or amount are not negative.
+
+public class ExchangeRate
+{
+    public decimal _rate;
+    public string FromCurrency { get; set; }
+    public string ToCurrency { get; set; }
+    public decimal Rate
+    {
+        get => _rate;
+        set
+        {
+            if (value < 0)
+            {
+                throw new Exception("Rate can't be Negative");
+            }
+            _rate = value;
+        }
+    }
+    public ExchangeRate(string fromCurrency, string toCurrency)
+    {
+        FromCurrency = fromCurrency;
+        ToCurrency = toCurrency;
+    }
+
+    public decimal Calculate(decimal amount)
+    {
+        if (amount < 0)
+        {
+            throw new Exception("Amount can't be Negative");
+        }
+        return Rate * amount;
+    }
+}
+
+/*
+Create interface IAnimal with property Name and Sound . 
+Create classes Cow, Cat and Dog that implement IAnimal . 
+Instantiate all three classes and pass them to a new method called MakeSound that has single parameter IAnimal 
+and it writes to console eg “Dog says woof woof” if instance of the Dog class is passed.
+*/
+public interface IAnimal
+{
+
+    string Name { get; }
+    string Sound { get; }
+    void MakeSound();
+
+}
+public class Cow : IAnimal
+{
+    public string Name { get => "Cow"; }
+    public string Sound { get => "Moo moo"; }
+    public void MakeSound()
+    {
+        System.Console.WriteLine($"{Name} says {Sound}");
+    }
+}
+
+public class Cat : IAnimal
+{
+    public string Name { get => "Cat"; }
+    public string Sound { get => "Meow meow"; }
+    public void MakeSound()
+    {
+        System.Console.WriteLine($"{Name} says {Sound}");
+    }
+}
+public class Dog : IAnimal
+{
+    public string Name { get => "Dog"; }
+    public string Sound { get => "Bhau bhau"; }
+    public void MakeSound()
+    {
+        System.Console.WriteLine($"{Name} says {Sound}");
+    }
+}
+
+/*
+Create Account class that can be initialized with the amount value. 
+Account class contains Withdraw and Deposit methods and Balance (get only) property. 
+Make sure that you can't withdraw more than you have in the balance.
+var account = new Account(100m);
+account.Deposit(100);
+Console.WriteLine($"Account balance is {account.Balance}");
+account.Withdraw(20);
+Console.WriteLine($"Account balance is {account.Balance}");
+account.Withdraw(200); // ❌ we should not be able to withdraw more than we have in the balance
+*/
+public class Account
+{
+    private double _balance;
+    public double Amount { get; set; }
+    public double Balance { get=>_balance; }
+  public Account(double amount)
+  {
+    this.Amount=amount;

```

Comment:
The text regarding initialization might be a bit misleading in the task. The initialized amount is basically a balance so assigning a balance to the amount would be correct in this case.